Source/Core/DiscIO/WIABlob.cpp

```diff
@@ -1702,6 +1702,10 @@ ConversionResultCode WIARVZFileReader<RVZ>::RunCallback(size_t groups_written, u
 
   const float completion = static_cast<float>(bytes_read) / iso_size;
 
+  // Progress Dialog will close prematurely if 100% is reached
+  if (completion >= 0.99f)
+    return callback(text, 0.99f) ? ConversionResultCode::Success : ConversionResultCode::Canceled;
```

Comment:
Could this be done in DolphinQt instead of in DiscIO? The behavior that you're trying to work around is in Qt, after all.

Reply:
I'll take a look at that. It doesn't seem to be a trivial fix though.